### PR TITLE
feat: Use personal api keys for throttling instead of teams

### PR DESCRIPTION
**Pull request by timgl**
## Problem

We throttle based on team, not personal api key. This is tricky because it means you can't separate out workloads for different types of calls and be sure that you won't be throttled by another job.

## Changes

Throttle based on personal api keys, not teams. To prevent abuse, implement a 10 api key/user limit.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
Very nice!

File: posthog/clickhouse/migrations/0073_update_person_batch_exports_query_for_memory.py
Comment (Twixes) [inline posthog/clickhouse/migrations/0073_update_person_batch_exports_query_for_memory.py:1]: What happened here? Not mentioned in the PR's description

**Review by Twixes — COMMENTED** [review]
Makes sense! Pushed a couple small commits for smoother edges ([diff](https://github.com/PostHog/posthog/pull/24039/files/9dbe5d2ed9b598d5f295fd39ff80699af2cf79b4..82336abda8d3990a9beed3954505e72c81714afc)).
However, no idea if `0073_update_person_batch_exports_query_for_memory.py` should be removed, since it seems completely unrelated here. @tomasfarias – got any context?

**Comment by Twixes** [comment]
Reopening just in case we still want to get this in – just the migration question remains @timgl

**Comment by timgl** [comment]
@Twixes reverted the migration, we're good to go now!

**Pull request merged**